enumerables.rb

```diff
@@ -0,0 +1,126 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    block_given? ? size.times { |i| yield(to_a[i]) } : (return to_enum)
+    self
+  end
+
+  def my_each_with_index
+    block_given? ? size.times { |i| yield(to_a[i], i) } : (return to_enum)
+    self
+  end
+
+  def my_select
+    arr = []
+    block_given? ? my_each { |e| arr.push(e) if yield(e) } : (return to_enum)
+    arr
+  end
+
+  def my_all?(arg = nil)
+    my_each do |e|
+      if block_given?
+        return false unless yield(e)
+      elsif !block_given? && !arg
+        return false
+      elsif arg.is_a? Regexp
+        return false unless e.to_s =~ arg
+      elsif arg.instance_of?(Class)
+        return false unless e.is_a? arg
+      elsif arg
+        return false unless e == arg
+      else
+        return true
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    my_each do |e|
+      if block_given?
+        return true if yield(e)
+      elsif !block_given? && !arg
+        return true if e
+      elsif arg.is_a? Regexp
+        return true if e.to_s =~ arg
+      elsif arg.instance_of?(Class)
+        return true if e.is_a? arg
+      elsif arg
+        return true if e == arg
+      else
+        return false
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    my_each do |e|
+      if block_given?
+        return false if yield(e)
+      elsif !block_given? && !arg
+        return true unless e
+      elsif arg.is_a? Regexp
+        return true unless e.to_s =~ arg
+      elsif arg.instance_of?(Class)
+        return true unless e.is_a? arg
+      elsif arg
+        return true unless e == arg
+      else
+        return false
+      end
+    end
+    false
+  end
+
+  def my_count(arg = nil)
+    if arg.nil? && !block_given?
+      size
+    elsif block_given?
+      counter = 0
+      my_each do |item|
+        counter += 1 if yield item
+      end
+      counter
+    else
+      arg_selected = my_select { |item| item == arg }
+      arg_selected.size
+    end
+  end
+
+  def my_map(proc = nil)
+    arr = []
+    return to_enum unless block_given?
+
+    if proc
+      my_each { |e| arr.push(proc.call(e)) }
+    else
+      my_each { |e| arr.push(yield(e)) }
+    end
+    arr
+  end
+
+  def my_inject(acc = nil)
+    my_each do |e|
+      if !acc
+        acc = e
+      elsif acc.instance_of?(Range)
+        my_each { |i| acc += i }
+      else
+        acc = yield(acc, e)
+      end
+    end
+    acc
+  end
+end
```

Comment:
- [ ] when a symbol is specified without an initial value, your my_inject should combine each element of the collection by applying the symbol as a named method. i.e `[1, 2, 3, 4].my_inject(:+)` should return 10.
- [ ]  when a symbol is specified without an initial value, your my_inject should combine each element of the collection by applying the symbol as a named method: range. i.e it should be able to work also on a range(`(1..4).my_inject(:+)` should return 10)
- [ ] when a symbol is specified with an initial value, my_inject combines each element of the collection by applying the symbol as a named method. i.e `[1, 2, 3, 4].my_inject(20, :*)` should return 480 and should also be able to work on a range too.